internal/controllers/core/dockercomposeservice/logs.go

```diff
@@ -0,0 +1,128 @@
+package dockercomposeservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/tilt-dev/tilt/internal/controllers/apicmp"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+)
+
+// Each DockerComposeService object owns a DockerComposeLogStream object of the same name.
+//
+// NOTE(nick): I think this might be the wrong API. Currently we model DockerCompose as two objects:
+//
+// 1) A service object with a Spec that tells us how to create the service and
+//    a Status that contains the container state
+// 2) A log object with a Spec that tells us how to watch logs.
+//
+// I think the better way to model this would be to more clearly separate "spinning up a service"
+// from "watching a service", so we'd actually have:
+//
+// 1) A service object with a Spec that tells us how to create the service and
+//    a Status that tells us whether the creation succeeded.
+// 2) A monitor object with a status that contains both the container state and the log state.
+//
+// But moving to this system will be easier once everything is in the API server.
+func (r *Reconciler) manageOwnedLogStream(ctx context.Context, nn types.NamespacedName, obj *v1alpha1.DockerComposeService) (reconcile.Result, error) {
```

Comment:
most of this logic is loosely adapted from KubernetesApply. The important bit is the comment on this function...not sure how strongly to go about changing this now, but seems like the better way to model this.

Reply:
Seems reasonable to me to hold off on for now - the split is very useful/important in K8s where there's multiple different ways to deploy but we want to share the monitoring logic across them. I can't imagine much of a demand for e.g. `dc_custom_deploy` though, so having them be a bit tighter coupled seems safe enough